source/common/router/router.cc

```diff
@@ -667,7 +671,8 @@ void Filter::onUpstreamReset(Http::StreamResetReason reset_reason,
   ENVOY_STREAM_LOG(debug, "upstream reset: reset reason {}", *callbacks_,
                    Http::Utility::resetReasonToString(reset_reason));
 
-  updateOutlierDetection(Http::Code::ServiceUnavailable, upstream_request);
+  updateOutlierDetection(Upstream::Outlier::Result::CONNECT_FAILED, upstream_request,
```

Comment:
In the non-split case don't you still need to track this as a 503? Or maybe that is done automatically?

Reply:
Done automatically in non-split mode:  https://github.com/cpakulski/envoy/blob/issue/3643/source/common/upstream/outlier_detection_impl.cc#L108-L109